src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,512 @@
+    const modalRef: NgbModalRef = this.ngbModal.open(IndividualExtensionConfirmModalComponent);
+    const studentsSelected = this.getSelectedStudents();
+    const instructorsSelected = this.getSelectedInstructors();
+    modalRef.componentInstance.studentsSelected = studentsSelected;
+    modalRef.componentInstance.instructorsSelected = instructorsSelected;
+    modalRef.componentInstance.extensionTimestamp = extensionTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.feedbackSessionTimeZone;
+    modalRef.componentInstance.modalType = ExtensionModalType.EXTEND;
+
+    modalRef.componentInstance.onConfirmExtensionCallBack.subscribe((isNotifyStudents: boolean) => {
+      this.handleCreateDeadlines(studentsSelected, instructorsSelected, extensionTimestamp, isNotifyStudents);
+      modalRef.close();
+    });
+  }
+
+  onDelete(): void {
+    const modalRef: NgbModalRef = this.ngbModal.open(IndividualExtensionConfirmModalComponent);
+    const studentsSelected = this.getSelectedStudents();
+    const instructorsSelected = this.getSelectedInstructors();
+    modalRef.componentInstance.modalType = ExtensionModalType.DELETE;
+    modalRef.componentInstance.studentsSelected = studentsSelected;
+    modalRef.componentInstance.instructorsSelected = instructorsSelected;
+    modalRef.componentInstance.extensionTimestamp = this.feedbackSessionEndingTime;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.feedbackSessionTimeZone;
+    modalRef.componentInstance.onConfirmExtensionCallBack.subscribe((isNotifyStudents: boolean) => {
+      this.handleDeleteDeadlines(studentsSelected, instructorsSelected, isNotifyStudents);
+      modalRef.close();
+    });
+  }
+
+  private handleCreateDeadlines(studentsSelected: StudentExtensionTableColumnModel[],
+                          instructorsSelected: InstructorExtensionTableColumnModel[],
+                          extensionTimestamp: number, isNotifyStudents: boolean): void {
+
+    const request: FeedbackSessionUpdateRequest = {
+      studentDeadlines: this.getUpdatedDeadlines(studentsSelected, extensionTimestamp),
+      instructorDeadlines: this.getUpdatedDeadlines(instructorsSelected, extensionTimestamp),
+      isGoingToNotifyAboutDeadlines: isNotifyStudents,
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService.updateFeedbackSession(this.courseId, this.feedbackSessionName, request)
+        .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+        .subscribe((updatedFeedbackSession: FeedbackSession) => {
+          this.studentDeadlines = updatedFeedbackSession.studentDeadlines ?? {};
+          this.instructorDeadlines = updatedFeedbackSession.instructorDeadlines ?? {};
+          this.statusMessageService.showSuccessToast(
+          `Extension for ${studentsSelected.length} students and ${instructorsSelected.length} instructors
+           have been successful!`);
```

Comment:
Maybe extension [...] has been, or extensions [...] have been successful?

Reply:
Yup, thanks for the catch